pay_ios/ios/Classes/PaymentHandler.swift

```diff
@@ -33,7 +38,7 @@ typealias PaymentCompletionHandler = (Bool) -> Void
 class PaymentHandler: NSObject {
   
   /// Holds the current status of the payment process.
-  var paymentStatus = PKPaymentAuthorizationStatus.failure
+  var paymentHandlerStatus: PaymentHandlerStatus = .started
```

Comment:
I'd suggest to use implicit unwrapping (`PaymentHandlerStatus!`) instead of initializing to an arbitrary value. Now that we are using `startPayment` to set the status to `.started` we can treat the declaration as a late initialization instead.